### PR TITLE
refactor(components/forms): selection box strict mode support

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/529/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/529/storybooks/ag-grid-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/529/storybooks/layout-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/529/storybooks/pages-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/529/storybooks/tabs-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/529/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/529/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/529/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/529/playground/)

File: apps/code-examples/src/app/code-examples/forms/selection-box/checkbox/selection-box-demo.component.ts [hunk 1]
```diff
@@ -6,9 +6,7 @@ import { FormArray, FormBuilder, FormGroup } from '@angular/forms';
   templateUrl: './selection-box-demo.component.html',
 })
 export class SelectionBoxDemoComponent implements OnInit {
-  public get checkboxArray(): FormArray {
-    return this.myForm.get('checkboxes') as FormArray;
-  }
+  public checkboxArray: FormArray;
```
Comment (Blackbaud-SteveBrush) [inline apps/code-examples/src/app/code-examples/forms/selection-box/checkbox/selection-box-demo.component.ts]: `FormArray | undefined`?

File: libs/components/forms/src/lib/modules/selection-box/selection-box-adapter.service.ts [hunk 2]
```diff
@@ -131,14 +133,17 @@ export class SkySelectionBoxAdapterService {
...
+    this.#renderer.addClass(nativeEl, newClass);
   }
 
   /**
    * Sets the `tabIndex` of the `element` to the provided `tabIndex`.
    */
-  public setTabIndex(element: ElementRef, tabIndex: number): void {
-    const el = element.nativeElement;
-    el.tabIndex = tabIndex;
+  public setTabIndex(element: ElementRef | undefined, tabIndex: number): void {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/src/lib/modules/selection-box/selection-box-adapter.service.ts]: This method doesn't make sense if we make the element `undefined`. Also, `tabIndex` is required, which comes after an optional parameter. Possible to revert this change and put the conditional logic on the caller's end?

File: libs/components/forms/src/lib/modules/selection-box/selection-box-grid.component.spec.ts [hunk 3]
```diff
@@ -104,29 +111,44 @@ describe('Selection box grid component', () => {
...
+    ).and.returnValue(300);
+    expect(resetHeightSpy).not.toHaveBeenCalled();
+    expect(syncMaxHeightSpy).not.toHaveBeenCalled();
 
     component.dynamicDescription = `Something really really really really really really really really really really really really really really really long to force a large height than before!`;
     fixture.detectChanges();
-    fixture.whenStable().then(() => {
-      expect(spy).toHaveBeenCalledTimes(1);
+    await fixture.whenStable();
+    setTimeout(() => {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/src/lib/modules/selection-box/selection-box-grid.component.spec.ts]: `setTimeout`? Should this `it` block use `fakeAsync`?

File: libs/components/forms/src/lib/modules/selection-box/selection-box-grid.component.ts [hunk 4]
```diff
@@ -49,122 +49,145 @@ export class SkySelectionBoxGridComponent
...
+      this.#updateChildrenHeights();
     }
   }
 
   @ViewChild('container', {
     read: ElementRef,
     static: true,
   })
-  private elementRef: ElementRef<any>;
+  public containerElementRef!: ElementRef<any>;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/src/lib/modules/selection-box/selection-box-grid.component.ts]: `containerElementRef: ElementRef | undefined;`

File: libs/components/forms/src/lib/modules/selection-box/selection-box-grid.component.ts [hunk 5]
```diff
@@ -49,122 +49,145 @@ export class SkySelectionBoxGridComponent
...
+  #mutationObserver: MutationObserver | undefined;
 
-  private mutationObserver: MutationObserver;
+  #ngUnsubscribe = new Subject<void>();
 
-  private ngUnsubscribe = new Subject<void>();
+  #_alignItems: SkySelectionBoxGridAlignItemsType = 'center';
 
-  private _alignItems: SkySelectionBoxGridAlignItemsType;
+  #_currentBreakpoint: SkyMediaBreakpoints | undefined;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/src/lib/modules/selection-box/selection-box-grid.component.ts]: This property is set to `| undefined` but its setter's definition is `set #currentBreakpoint(value: SkyMediaBreakpoints)`. Is it possible for this property to be set to undefined? Should it have a default value instead?

File: apps/code-examples/src/app/code-examples/forms/selection-box/checkbox/selection-box-demo.component.ts (hunk 1 above)
Comment (Blackbaud-TrevorBurch) [reply apps/code-examples/src/app/code-examples/forms/selection-box/checkbox/selection-box-demo.component.ts]: Done

File: libs/components/forms/src/lib/modules/selection-box/selection-box-grid.component.spec.ts (hunk 3 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/forms/src/lib/modules/selection-box/selection-box-grid.component.spec.ts]: So I actually met with Erika about this - happy to get on a call with you if needed. This test was suffering from an issue where the mutation observer isn't firing in time and the test is failing. We tried a ton of combinations of `fakeAsync` and `tick` calls and standard async with `whenStable` and/or `whenRenderingDone` calls and the only thing that we could get to work consistently was adding the `setTimeout`. Happy to chat with you though.

File: libs/components/forms/src/lib/modules/selection-box/selection-box-grid.component.ts (hunk 5 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/forms/src/lib/modules/selection-box/selection-box-grid.component.ts]: I made a change around this to give a default. I almost prefer the `| undefined` though as we have to call through the setter anyway to set the default classes and just letting the `ngAfterViewInit` handle this with it just being `undefined` prior seems more efficient. Thoughts?

File: libs/components/forms/src/lib/modules/selection-box/selection-box-adapter.service.ts (hunk 2 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/forms/src/lib/modules/selection-box/selection-box-adapter.service.ts]: Good call. I think that was from an earlier attempt at things and I missed rolling this one back. I have reverted.

File: libs/components/forms/src/lib/modules/selection-box/selection-box-grid.component.spec.ts (hunk 3 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/forms/src/lib/modules/selection-box/selection-box-grid.component.spec.ts]: We talked. Made a change.

File: libs/components/forms/src/lib/modules/selection-box/selection-box-grid.component.ts (hunk 4 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/forms/src/lib/modules/selection-box/selection-box-grid.component.ts]: Fixed

File: libs/components/forms/src/lib/modules/selection-box/selection-box-grid.component.ts (hunk 5 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/forms/src/lib/modules/selection-box/selection-box-grid.component.ts]: Fixed

**Review by Blackbaud-SteveBrush — APPROVED** [review]
